packages/paste-core/components/pagination/src/PaginationArrow/index.tsx

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+import {PaginationBackArrow} from './PaginationBackArrow';
+import {PaginationForwardArrow} from './PaginationForwardArrow';
+import type {PaginationArrowProps} from '../types';
+import {PaginationArrowPropTypes} from '../proptypes';
+import {handleLinkValidation} from '../utils';
+
+const PaginationArrowVariants = {
+  back: PaginationBackArrow,
+  forward: PaginationForwardArrow,
+};
+
+const PaginationArrow = React.forwardRef<HTMLButtonElement, PaginationArrowProps>(
+  ({as = 'button', href, variant = 'back', ...props}, ref) => {
+    handleLinkValidation({as, href});
+    const PaginationArrowComponent = PaginationArrowVariants[variant];
+    return <PaginationArrowComponent {...props} ref={ref} as={as} href={href} />;
```

Comment:
It's probably a little cleaner to just do the conditional in here without componentizing arrow back/forward, like:

```js
const VisibleLabel = ({visibleLabel, ...props}) => {
  if (!visibleLabel) return null;
  return (
    <Text aria-hidden="true" as="span" color="inherit" {...props}>
      {visibleLabel}
    </Text>
  );
}

....



<PaginationArrowButton {...props} ref={ref} as={as} visibleLabel={visibleLabel}>
    {variant === "forward"  ? <VisibleLabel marginRight="space30" label={visibleLabel} /> : null}
    <Box
      alignItems="center"
      borderColor="inherit"
      borderRadius="borderRadiusCircle"
      borderStyle="solid"
      borderWidth="borderWidth10"
      display="flex"
      height="sizeSquare80"
      justifyContent="center"
      width="sizeSquare80"
    >
       {variant === "forward" ? <ArrowForwardIcon decorative={false} display="block" title={label} /> : ...}
    </Box>
     {variant === "back"  ? <VisibleLabel marginLeft="space30" label={visibleLabel} /> : null}
</PaginationArrowButton>
```

Reply:
I restructured this slightly to make it more clear which variant is being rendered. I kept the original basic idea here because the conditionals throw me off a bit. I like having the two separate variant components. For me, it makes it easier to see whats going on the files. Also, if anything needs to change I know exactly where to make that change.